cloudify_agent/api/pm/base.py

```diff
@@ -249,8 +249,7 @@ def __init__(self, logger=None, **params):
                                         defaults.LOG_FILE_SIZE)
         self.log_max_history = params.get('log_max_history',
                                           defaults.LOG_BACKUPS)
-        self.executable_temp_path = params.get('executable_temp_path') or \
-            get_exec_tempdir()
+        self.executable_temp_path = params.get('executable_temp_path')
```

Comment:
This is needed because of a bug that's been hidden so far. If the installation method is `init_script`, then this code on Windows is always executed as `Administrator`. If `params.get('executable_temp_path')` returns `None`, then the template will end up receiving `Administrator`'s tempdir (under `C:\Users\Administrator\AppData\...`).

This means that, when this agent creates temporary files to run scripts, it will try to create them in `Administrator`'s tempdir. It works fine when `LOCALSYSTEM` is running the service, but fails when the user running the service has no permissions to access `Administrator`'s tempdir.

The correct way to approach this is to not render `CFY_EXEC_TEMP` at all, if it's not explicitly defined.

This fix is complemented with a fix (see below) to `nssm.conf.template`, surrounding the rendering of `CFY_EXEC_TEMP` with a Jinja `if` statement.

Reply:
So if this is set to None, where do temp files end up when the agent creates them?

Reply:
Oh, I just saw the comment below, ignore this.